stdlib/Dates/test/io.jl

```diff
@@ -545,4 +554,13 @@ end
     @test Time("13:24", "II:MMp") == Time("13:24", "HH:MM")
 end
 
+@testset "Issue #10561, two-digit year parsing ambiguities" begin
+    for test_year in [0, 1, 100, 101, 1000, 1001, 1900, 1901, 2000, 2001, 2100, 2101]
+        fmt_string = dateformat"yy"
+
+        # All dates encoded in two digit year format YY are parsed as year 00YY
+        @test Date(Dates.format(Date(test_year), fmt_string), fmt_string) == Date(test_year % 100)
+    end
```

Comment:
Looking over this again I realized that these tests are slightly misleading. As we're testing with just two digit dates the tests for `1`, `101`, `1001`, etc. are all identical. The point of these tests is about parsing not about formatting right? Note that parsing a year string with more than two-digits works as it should (e.g. `Date("2101", dateformat"yy") == Date(2101,1,1)`).

With that in mind:
```suggestion
    # All dates encoded in two digit year format YY are parsed as year 00YY
    for test_year in [0, 1, 99, 2101]
        @test Date(lpad(test_year, 2, '0'), dateformat"yy") == Date(test_year)
    end
```